src/components/AttachmentModal.js

```diff
@@ -274,7 +274,7 @@ class AttachmentModal extends PureComponent {
                         title={this.props.headerTitle || this.props.translate('common.attachment')}
                         shouldShowBorderBottom
                         shouldShowDownloadButton={this.props.allowDownload}
-                        onDownloadButtonPress={() => this.downloadAttachment(source)}
+                        onDownloadButtonPress={() => this.downloadAttachment(this.state.source)}
```

Comment:
`source` from here was previously set to `this.state.source` and the download functionality was working properly. By updating this, after I was downloading an image, I couldn't open it locally because it was wrongly downloaded. Updated it back to fix this & updated the QA steps with this use case.

Reply:
Trying to open the file after downloading it shows me this:
![image](https://github.com/Expensify/App/assets/56603839/002031db-6694-42fa-a0d3-57552d4c7745)